src/test/run-pass/zero-sized-btreemap-insert.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2013 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::cmp::{Ord, Ordering, PartialOrd};
+use std::collections::BTreeMap;
+use std::iter::Iterator;
+
+#[derive(Eq, Hash, Debug, Ord)]
+struct Zst;
+
+impl PartialEq for Zst {
+    fn eq(&self, rhs: &Self) -> bool { false }
+}
+
+impl PartialOrd for Zst {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> { Some(Ordering::Less) }
+}
+
+fn main() {
+    const N: usize = 8;
+
+    // Test that for all possible sequences of push,
+    // we end up with a BTreeMap of the correct size
+
+    for len in 0..N {
+        let mut tester = BTreeMap::new();
+        assert_eq!(tester.len(), 0);
+        for case in 0..(1 << len) {
+            assert_eq!(tester.len(), 0);
+            for bit in 0..len {
+                tester.insert(Zst, ());
+            }
+            assert_eq!(tester.len(), len);
+            assert_eq!(tester.iter().count(), len);
+            tester.clear();
```

Comment:
This test is incorrect, you'll never end up increasing the length past 1. It's a map, not a multi-map.